### PR TITLE
feat(facebook-to-local): allow users to move from facebook=>local

### PR DESCRIPTION
**Pull request by lefnire**
Allows users to move from Facebook to Local auth.
1. Option to add Local Auth
2. Option de-register Facebook Auth, only after you've added Local Auth

### CODE REVIEW

**Comment by lefnire** [comment]
https://github.com/HabitRPG/habitrpg-shared/pull/464

**Comment by lefnire** [comment]
Tested locally, but wanna bounce off anyone (@Alys @crookedneighbor @paglias et al) to take a peek just in case. Will deploy tonight otherwise, I have 1.5 shit-tons of "Facebook=>Local" emails stock-piled for this puppy

File: src/controllers/auth.js [hunk 1]
```diff
@@ -190,9 +192,18 @@ api.loginSocial = function(req, res, next) {
 
 /**
  * DELETE /user/auth/social
- * TODO implement
  */
-api.deleteSocial = function(req,res,next){next()}
+api.deleteSocial = function(req,res,next){
+  if (!res.locals.user.auth.local.username)
+    return res.json(401, {err:"Account lacks another authentication method, can't detach Facebook"});
+  //FIXME for some reason, the following gives https://gist.github.com/lefnire/f93eb306069b9089d123
```
Comment (crookedneighbor) [inline src/controllers/auth.js:199]: Weird. What do you think this is about?

Comment (Alys) [inline src/controllers/auth.js:199]: dayStart will likely be referring to `preferences.dayStart` (custom day
start) from the user's object, so `preferences` isn't available at this
point, or the entire user object isn't. Apologies if you knew that and were
wanting significantly more useful information. :)

Comment (lefnire) [inline src/controllers/auth.js:199]: Yeah, `user` (or `this` in the `User.on('save')` hook) is a completely empty object. I have no idea why, I haven't encountered that before..

**Pull request merged**